### ISSUE TITLE
Delegate cop should ignore class delegation

### ISSUE THREAD

**Issue by mockdeep** [bbatsov/rubocop#1509]
We have code like this:

```ruby
class MyClass

  def self.do_something(args)
    new.do_something(args)
  end

  def do_something(args)
    ...
  end
end
```

And we get a delegate violation: `Rails/Delegate: Use delegate to define delegations.`.

**Comment by ivanoblomov** [comment]
+1

**Comment by EsterYtterbrinkAtTrialbee** [comment]
+1

**Comment by walterhorstman** [comment]
I've just stumbled over the same issue when I write the following code. If the code is wrong, please tell me how to fix it. Else, it might be a new bug, or this one should be reopened.

``` ruby
def self.[](code)
  indexed[code]
end

def self.indexed
  all.index_by(&:code)
end
```

My goal is to be able to write things like `MyClass[:something]` instead of `MyClass.indexed[:something]`.

**Comment by bbatsov** [comment]
This is fixed on `master`.

**Comment by walterhorstman** [comment]
Are you sure? I've changed the definition in the Gemfile to use the Github repo and it still occurs.

**Comment by bbatsov** [comment]
I'm sure we're no longer processing class methods, so I cannot imagine how this can still happen.
